plugins/collaajUpdatePlugin/lib/model/collaajini.php

```diff
@@ -0,0 +1,96 @@
+<?php
+
+class collaajini
+{
+    const INI_FILE_NAME = "../../collaajClientUpdate.ini";
+    const AVAILABLE_OS_TYPES = "Mac OS X,Windows";
+    const UPDATE_KEY = "version";
+    const DOWNLOAD_URL = "download_url";
+    const MD5 = "md5";
+    const DOWNLOAD_URL_TEMPLATE = "http://<HOSTNAME>/api_v3/index.php/service/collaajupdate_collaajupdate/action/<ACTION>/os/<OS>/version/<VERSION>/filename/";
+
+    private $collaaj_data = "";
+
+    private $returned_data = array(
+        "os" => "",
+        "version" => "",
+        "download_url" => "",
+        "md5" => "",
+    );
+
+    public function __construct() {
+        $relative_ini_path = dirname(__FILE__) .'/'.self::INI_FILE_NAME;
+        $this->collaaj_data = new collaajInstallData($relative_ini_path);
+    }
+
+    public function returnLatestVersionUrl($given_os, $current_version) {
+        $actual_extension = $this->identifyNeededExtension($given_os, "install");
+        $ini_returned_data = $this->collaaj_data->getLatestVersionData($actual_extension);
+        if ($ini_returned_data) {
+            foreach ($ini_returned_data as $key => $value) {
+                $this->returned_data[$key] = $value;
+            }
+            // Appending the current file name to the to future url
+            $this->returned_data["url"] = $this->prepareDownloadUrl($given_os,$current_version, "serveinstall",self::DOWNLOAD_URL_TEMPLATE.$this->returned_data[self::DOWNLOAD_URL]);
+            return $this->returned_data[self::DOWNLOAD_URL];
+        } else throw new KalturaAPIException ("No install version found for $given_os");
+    }
+
+    public function returnUpdateFileUrl($given_os, $current_version) {
+        $actual_extension = $this->identifyNeededExtension($given_os, "update");
+        $returned_data = $this->collaaj_data->returnVersionUpdateInfo($actual_extension, $current_version);
+        if ($returned_data != 1) {
+            foreach ($returned_data as $key => $value) {
+                $this->returned_data[$key] = $value;
+            }
+            $this->returned_data["url"] = $this->prepareDownloadUrl($given_os,$current_version, "serveupdate",self::DOWNLOAD_URL_TEMPLATE.$this->returned_data[self::DOWNLOAD_URL]);
+            return $this->returned_data;
+
+        } else throw new KalturaAPIException ("There is no update for $current_version");
+    }
+
+    public function prepareDownloadUrl($needed_os, $current_version, $action, $original_url) {
+        $returned_url = $original_url;
+        $returned_url = str_replace ("<HOSTNAME>", gethostname(),$returned_url);
```

Comment:
Hostname means nothing beyond firewall or load-balancer, use the common functionality to return URLs to API calls, see getURL of other serve actions.